src/harness/fourslashImpl.ts

```diff
@@ -4510,6 +4510,58 @@ export class TestState {
 
         this.verifyCurrentFileContent(newFileContent);
     }
+
+    public baselineMapCode(
+        ranges: Range[],
+        changesFilename: string,
+    ): void {
+        const changes = this.getFileContent(changesFilename);
+        let fileName: string | undefined;
+        const focusLocations = ranges.map(({ fileName: fn, pos, end }) => {

```

Comment:
instead of sorting these ranges here can you make sure tests pass in required focus locations and they are passed as is to LS so that tests can be more explicit on order they are sending the ranges in.

Reply:
Also the ranges need to be optional and the here should take a file name so that you can test when there are no focus locations (they are marked optional) - Again test for the same pls.

Reply:
I'll write a test with no ranges, but the sorting is not for testing functionality, but for properly outputting the baselines. It's irrelevant to the test.